solidity/contracts/core/AutomationVault.sol

```diff
@@ -33,9 +34,15 @@ contract AutomationVault is IAutomationVault {
   mapping(address _relay => EnumerableSet.AddressSet _jobs) internal _approvedJobs;
 
   /**
-   * @notice Selectors approved for a job to be executed by a relay
+   * @notice Selectors approved to be executed by a relay for a specific job and type
    */
-  mapping(address _relay => mapping(address _job => EnumerableSet.Bytes32Set _selectors)) internal _approvedJobSelectors;
+  mapping(address _relay => mapping(address _job => mapping(bytes4 _selector => IAutomationVault.HookData))) internal
+    _approvedJobSelectorsWithHooks;
+
+  /**
+   * @notice List of approved relays
+   */
+  mapping(address _relay => mapping(address _job => bytes4[] _selectors)) internal _approvedJobSelectorsList;
 
```

Comment:
wait, you're loosing the `approvedJobSelectors.contains(selector)`, how are you accessing it now? to check if a selector was approved